src/content/en/2021/third-parties.md

```diff
@@ -1,20 +1,491 @@
+
+The first non-Google entity is Facebook with a comparatively low usage of 29.18%. This is followed by Cloudflare's CDN fronting popular libraries and other resources (despite being listed as amp.cloudflare.com it also includes the much larger cdnjs.cloudflare–this has been updated to show the more commonly used domain for next year).
+
+After this we're back to Google with YouTube, and Maps two spots later. The remaining spots are filled with CDNs for other popular libraries and tools.
+
+## Performance impact of third parties
+
+Using third parties can have a noticeable impact on performance. That's not necessarily a consequence of them being a third party per se—the same functionality implemented by a site owner as a first-party resource would often cause similar performance issues—if not worse, given the expertise the third party should have on the particular field.
+
+So it's not the fact they are third-party resources, so much as what those resources are doing. And most third-party usage depends on the third-party service, rather than just as a place to serve it from.
+
+However, a third party's business is in allowing their content or service to be hosted on many websites, they have a duty to ensure they minimise the negative impact of that. Especially since site owners often have limited control over, and influence on the performance impact of third parties other than to use them or not.
+
+
+### Using third-party domains versus self-hosting
+
+There is a definite cost to connecting to another domain, even though most third parties will be using globally distributed, high-performance CDNs, and many web performance advocates (including myself!) recommend self-hosting where possible to avoid this penalty. This is particularly relevant now that all the major browsers have moved away from sharing caches between origins, so the claim that once one site has downloaded that resource, others can benefit from it is no longer true. Though this was a questionable claim even in the past in this author's opinion given the number of versions of libraries, and limitations of the HTTP cache.
+
+Saying that, rarely is life as definitive as we would like and, in some cases self-hosting may actually cost performance. I've <a hreflang="en" href="https://www.tunetheweb.com/blog/should-you-self-host-google-fonts/">written before how the question on whether to self-host Google Fonts</a> is not as clear cut as it might seem and requires a degree of expertise to ensure you are replicating all that Google Fonts does for you in the performance front. To avoid that hassle you can just use the hosted version, and ensure you're <a hreflang="en" href="https://csswizardry.com/2020/05/the-fastest-google-fonts/">reducing the performance impact as much as possible</a>.
```

Comment:
Changed to:

```html
To avoid that hassle you can just use the hosted version, and ensure you're reducing the performance impact as much as possible, as discussed by <a hreflang="en" href="https://twitter.com/csswizardry">Harry Roberts</a> in his <a hreflang="en" href="https://csswizardry.com/2020/05/the-fastest-google-fonts/">The Fastest Google Fonts</a> post.
```